Unable to Convert Triggers with Complex Responses

In attempting to convert the following trigger to TypeScript:
```ts
{
  id: 'ShivaEx Sword Phase',
  type: 'Ability',
  netRegex: NetRegexes.ability({ source: 'Shiva', id: '993', capture: false }),
  netRegexDe: NetRegexes.ability({ source: 'Shiva', id: '993', capture: false }),
  netRegexFr: NetRegexes.ability({ source: 'Shiva', id: '993', capture: false }),
  netRegexJa: NetRegexes.ability({ source: 'シヴァ', id: '993', capture: false }),
  netRegexCn: NetRegexes.ability({ source: '希瓦', id: '993', capture: false }),
  netRegexKo: NetRegexes.ability({ source: '시바', id: '993', capture: false }),
  response: (data, _matches, output) => {
    // cactbot-builtin-response
    output.responseOutputStrings = {
      swordTankSwap: {
        en: 'Sword (Tank Swap)',
        de: 'Schwert (Tankwechsel)',
        fr: 'Épée (Tank Swap)',
        ja: '剣 (スイッチ)',
        cn: '剑（换T）',
        ko: '검 (탱커 교대)',
      },
      sword: {
        en: 'Sword',
        de: 'Schwert',
        fr: 'Épée',
        ja: '剣',
        cn: '剑',
        ko: '검',
      },
    };
    if (data.role === 'tank') {
      if (data.currentTank && data.slashing && data.slashing[data.currentTank])
        return { alertText: output.swordTankSwap!() };
    }

    return { infoText: output.sword!() };
  },
  run: (data) => data.soonAfterWeaponChange = true,
},
```
We receive this output:
```sh
$ tsc
ui/raidboss/data/02-arr/trial/shiva-ex.ts(116,7): error TS2322: Type '(data: Data, _matches: Matches & ToMatches<NetAbilityFields>, output: Output) => { alertText: string; infoText?: undefined; } | { ...; }' is not assignable to type 'ResponseField<Data, Matches> | ResponseField<Data, Matches & ToMatches<NetAbilityFields>> | ... 21 more ... | undefined'.
  Type '(data: Data, _matches: Matches & ToMatches<NetAbilityFields>, output: Output) => { alertText: string; infoText?: undefined; } | { ...; }' is not assignable to type 'ResponseFunc<Data, Matches>'.
    Types of parameters '_matches' and 'matches' are incompatible.
      Type 'Matches' is not assignable to type 'Matches & ToMatches<NetAbilityFields>'.
        Type 'Matches' is not assignable to type 'ToMatches<NetAbilityFields>'.
```

--- a/types/trigger.d.ts
+++ b/types/trigger.d.ts
@@ -1,5 +1,5 @@
 import { Lang, NonEnLang } from '../resources/languages';
-import { NetMatches, Matches } from '../types/net_matches';
+import { NetMatches, NetAnyMatches } from '../types/net_matches';
 import { TimelineReplacement, TimelineStyle } from '../ui/raidboss/timeline';
 
 import { RaidbossData } from './data';
@@ -35,31 +35,31 @@ export type Output = {
 };
 
 // The output of any non-response raidboss trigger function.
-export type TriggerOutput<Data extends RaidbossData, MatchType extends Matches> =
+export type TriggerOutput<Data extends RaidbossData, MatchType extends NetAnyMatches> =
     undefined | null | LocaleText | string | number | boolean |
     ((d: Data, m: MatchType, o: Output) => TriggerOutput<Data, MatchType>);
 
 // Used if the function doesn't need to return an en key
-export type PartialTriggerOutput<Data extends RaidbossData, MatchType extends Matches> =
+export type PartialTriggerOutput<Data extends RaidbossData, MatchType extends NetAnyMatches> =
     undefined | null | Partial<LocaleText> | string | number | boolean |
     ((d: Data, m: MatchType, o: Output) => PartialTriggerOutput<Data, MatchType>);
 
 // The type of a non-response trigger field.
-export type TriggerFunc<Data extends RaidbossData, MatchType extends Matches, Return> =
+export type TriggerFunc<Data extends RaidbossData, MatchType extends NetAnyMatches, Return> =
     (data: Data, matches: MatchType, output: Output) => Return;
 
 // The output from a response function (different from other TriggerOutput functions).
-export type ResponseOutput<Data extends RaidbossData, MatchType extends Matches> = {
+export type ResponseOutput<Data extends RaidbossData, MatchType extends NetAnyMatches> = {
   infoText?: TriggerField<Data, MatchType, TriggerOutput<Data, MatchType>>;
   alertText?: TriggerField<Data, MatchType, TriggerOutput<Data, MatchType>>;
   alarmText?: TriggerField<Data, MatchType, TriggerOutput<Data, MatchType>>;
   tts?: TriggerField<Data, MatchType, PartialTriggerOutput<Data, MatchType>>;
 } | undefined;
 // The type of a response trigger field.
-export type ResponseFunc<Data extends RaidbossData, MatchType extends Matches> =
+export type ResponseFunc<Data extends RaidbossData, MatchType extends NetAnyMatches> =
     (data: Data, matches: MatchType, output: Output) => ResponseOutput<Data, MatchType>;
 
-export type ResponseField<Data extends RaidbossData, MatchType extends Matches> =
+export type ResponseField<Data extends RaidbossData, MatchType extends NetAnyMatches> =
     ResponseFunc<Data, MatchType> | ResponseOutput<Data, MatchType>;
 
 export type TriggerAutoConfig = {
@@ -74,7 +74,7 @@ export type TriggerAutoConfig = {
 
 // Note: functions like run or preRun need to be defined as void-only as (confusingly)
 // it is not possible to assign `(d: Data) => boolean` to a void | undefined, only to void.
-export type TriggerField<Data extends RaidbossData, MatchType extends Matches, Return> =
+export type TriggerField<Data extends RaidbossData, MatchType extends NetAnyMatches, Return> =
   [Return] extends [void] ? TriggerFunc<Data, MatchType, void> :
   TriggerFunc<Data, MatchType, Return | undefined> | Return | undefined;
 

